testsuite/tests/typing-gadts/pr10735.ml

```diff
@@ -20,11 +20,8 @@ type (_, _) eq = Refl : ('a, 'a) eq
 let () =
   let (Refl : (bool X.t, bool t) eq) as t = Obj.magic  () in ()
 [%%expect{|
-Line 2, characters 7-11:
+Line 2, characters 6-41:
 2 |   let (Refl : (bool X.t, bool t) eq) as t = Obj.magic  () in ()
-           ^^^^
-Error: This pattern matches values of type (bool X.t, bool X.t) eq
-       but a pattern was expected which matches values of type
-         (bool X.t, bool t) eq
-       Type bool X.t is not compatible with type bool t
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Warning 26 [unused-var]: unused variable t.
```

Comment:
I wish the `in ()` part at the end was rewritten into `in ignore t` to silence the warning.